apps/frontend/src/modules/auth/components/PrivateRoute.tsx

```diff
@@ -0,0 +1,237 @@
+import { GlobalSpinner, NotFound } from "@/components";
+import { createOrgDbs } from "@/database";
+import {
+    OrganizationCountableEdge,
+    useViewerLazyQuery,
+} from "@/generated/graphql";
+import { handleRedirect } from "@/modules/auth/helper";
+import { useAuthToken } from "@/modules/auth/hooks/useAuthToken";
+import { useSession } from "@/modules/users/hooks/useSession";
+import { SessionViewer } from "@/types";
+import { omitTypename } from "@/utils";
+import { logDebug } from "@/utils/log";
+import { useEffect, useState } from "react";
+import { Navigate, useNavigate, useParams } from "react-router-dom";
+
+type Props = {
+    children: React.ReactNode;
+};
+
+export const PrivateRoute = ({ children }: Props) => {
```

Comment:
@AfeezGL I think we should move both PublicRoute and PrivateRoute components in `layout` directory.